src/core/gui/inputdevices/GeometryToolInputHandler.cpp

```diff
@@ -0,0 +1,335 @@
+                }
+                break;
+            }
+            case GDK_KEY_m: {
+                controller->markPoint(translationX, translationY);
+                return true;
+            }
+        }
+
+        if (xdir != 0 || ydir != 0) {
+            const auto c = std::cos(rotation);
+            const auto s = std::sin(rotation);
+            double xshift{0.0};
+            double yshift{0.0};
+            const auto amount = (event.state & GDK_MOD1_MASK) ? MOVE_AMOUNT_SMALL : MOVE_AMOUNT;
+            if (event.state & GDK_SHIFT_MASK) {
+                xshift = amount * (c * xdir - s * ydir);
+                yshift = amount * (s * xdir + c * ydir);
+            } else {
+                xshift = amount * xdir;
+                yshift = amount * ydir;
+            }
+            controller->move(xshift, yshift);
+            return true;
+        }
+
+        auto p = utl::Point(translationX, translationY);
+        if (angle != 0) {
+            controller->rotate(angle, p.x, p.y);
+            return true;
+        }
+        if (scale != 1.0) {
+            controller->scale(scale, p.x, p.y);
+            return true;
+        }
+    }
+    return false;
+}
+
+void GeometryToolInputHandler::sequenceStart(InputEvent const& event) {
+    if (event.sequence == this->primarySequence) {
+        this->priLastPageRel = this->getCoords(event);
+    } else {
+        this->secLastPageRel = this->getCoords(event);
+    }
+    const auto page = controller->getPage();
+
+    const auto l = page->getSelectedLayer();
+    this->lines.clear();
+    for (const auto& e: l->getElements()) {
+        if (e->getType() == ELEMENT_STROKE) {
+            auto s = dynamic_cast<Stroke*>(e);
+            if (s->getPointCount() == 2) {
+                lines.push_back(s);
+            }
+        }
+    }
+}
+
+void GeometryToolInputHandler::scrollMotion(InputEvent const& event) {
+    // Will only be called if there is a single sequence (zooming handles two sequences)
+    auto offset = [&]() {
+        auto coords = this->getCoords(event);
+        if (event.sequence == this->primarySequence) {
+            const auto offset = coords - this->priLastPageRel;
+            this->priLastPageRel = coords;
+            return offset;
+        } else {
+            const auto offset = coords - this->secLastPageRel;
+            this->secLastPageRel = coords;
+            return offset;
+        }
+    }();
+    const auto pos = Point(translationX + offset.x, translationY + offset.y);
+    double minDist = SNAPPING_DISTANCE_TOLERANCE;
+    const auto angleSetsquare = rotation;
+    double diffAngle{NAN};
+    for (const auto& l: lines) {
+        auto first = l->getPoint(0);
+        auto second = l->getPoint(1);
+        auto dist = Snapping::distanceLine(pos, first, second);
+        auto angleLine = std::atan2(second.y - first.y, second.x - first.x);
+        auto diff = std::remainder(angleLine - angleSetsquare, M_PI_2);
```

Comment:
I think to many `auto`'s make the code harder to understand.

Reply:
Ok, I replaced some `auto`s throughout this file.